src/components/molecules/switch.tsx

```diff
@@ -0,0 +1,136 @@
+import * as React from "react";
+import { useAnimation, motion } from "framer-motion";
+import { Box, BoxProps, useColorMode } from "@chakra-ui/core";
+
+const MotionBox = motion.custom(Box);
+
+// TypeScript ‘type alias’
+type Props = Partial<BoxProps> &
+  Partial<{
+    // Optional props
+    setCurrentBy: boolean;
+    currentIndex: number;
+    currentOption: string;
+    onValueChange: (option: string, index: number) => void;
+  }> & {
+    // Required props
+    options: string[];
+  };
+
+// passing the `Props` type alias to the component
+export function SegmentedControl(props: Props) {
+  // Splitting up the props
+  const {
+    width,
+    setCurrentBy,
+    currentIndex,
+    options,
+    currentOption,
+    onValueChange,
+  } = props;
+
+  // The initial selected index is set by prop.currentIndex or prop.currentOption
+  // (depending on which was selected with props.setCurrentBy)
+  let initialIndex;
+  if (setCurrentBy === true) {
+    initialIndex = currentIndex;
+  } else {
+    initialIndex = options.indexOf(currentOption);
+  }
+  const [index, setIndex] = React.useState(initialIndex); // The ‘index’ state
+
+  // An AnimationControls object for the Selection Indicator
+  const selectionIndicator = useAnimation();
+
+  // Calculating the width of each segment
+  const widthSegment =
+    ((width as number) - (options.length - 1)) / options.length;
+
+  // Change selection function
+  const changeSelectedIndex = (newIndex) => {
+    // Move the Selection Indicator
+    selectionIndicator.start({ x: newIndex * widthSegment + newIndex * 1 });
+
+    // Trigger the event property
+    onValueChange(options[newIndex], newIndex);
+    // Update the index state
+    setIndex(newIndex);
```

Comment:
Here is where you'd need to call the `setIndex` from `[testId].tsx` in addition to the local `setIndex`. As I mention in the comment below, if possible, try to consolidate them as it seems like their logic is tightly coupled, although if it's not possible that's fine.